File: android-app/app/src/main/java/com/example/campusconnect/map/MapMarker.java
package com.example.campusconnect.map;

public class MapMarker {

    private MarkerType type;
    private double latitude;
    private double longitude;
    private String label;

    private float x;
    private float y;

    public MapMarker(MarkerType type, double latitude, double longitude, String label) {
        this.type = type;
        this.latitude = latitude;
        this.longitude = longitude;
        this.label = label;
    }

    public MarkerType getType() {
        return type;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public String getLabel() {
        return label;
    }

    public float getX() {
        return x;
    }

    public float getY() {
        return y;
    }

    public void setPixelPosition(float x, float y) {
        this.x = x;
        this.y = y;
    }
}

File: android-app/app/src/main/java/com/example/campusconnect/feature/map/mapengine/MapMarker.kt
package com.example.campusconnect.feature.map.mapengine

data class MapMarker(
    val id: Int,                         // Strict Int ID (Matches Backend Entity Primary Key)
    val sourceId: Int? = null,           // Strict Int ID
    val type: MarkerType,
    val latitude: Double,
    val longitude: Double,
    val label: String,

    // Backend Fields
    val userId: Int? = null,             // Strict Int
    val eventId: Int? = null,            // Strict Int
    val insideCampus: Boolean? = null,   // Campus presence badge

    // UI Visual & Render Fields
    val gender: String? = null,
    val size: MarkerSize = MarkerSize.MEDIUM,
    val isHighlighted: Boolean = false,
    var x: Float = 0f,
    var y: Float = 0f,
    val state: MarkerState = MarkerState.DEFAULT,
    val priority: Int = 0,
    val isActive: Boolean = true
)

enum class MarkerState {
    DEFAULT,
    SELECTED,
    ACTIVE,
    DISABLED
}

enum class MarkerSize {
    SMALL,
    MEDIUM,
    LARGE
}